worlds/sc2wol/MissionTables.py

```diff
@@ -171,40 +287,101 @@ class FillMission(NamedTuple):
+        SC2Mission.ALL_IN.mission_name: MissionInfo(SC2Mission.ALL_IN, [MissionConnection(23, SC2Campaign.WOL), MissionConnection(24, SC2Campaign.WOL)], SC2Mission.ALL_IN.area, or_requirements=True, completion_critical=True)
+    },
+    SC2Campaign.PROPHECY: {
+        SC2Mission.WHISPERS_OF_DOOM.mission_name: MissionInfo(SC2Mission.WHISPERS_OF_DOOM, [MissionConnection(9, SC2Campaign.WOL)], SC2Mission.WHISPERS_OF_DOOM.area),
+        SC2Mission.A_SINISTER_TURN.mission_name: MissionInfo(SC2Mission.A_SINISTER_TURN, [MissionConnection(1, SC2Campaign.PROPHECY)], SC2Mission.A_SINISTER_TURN.area),
+        SC2Mission.ECHOES_OF_THE_FUTURE.mission_name: MissionInfo(SC2Mission.ECHOES_OF_THE_FUTURE, [MissionConnection(2, SC2Campaign.PROPHECY)], SC2Mission.A_SINISTER_TURN.area),
+        SC2Mission.IN_UTTER_DARKNESS.mission_name: MissionInfo(SC2Mission.IN_UTTER_DARKNESS, [MissionConnection(3, SC2Campaign.PROPHECY)], SC2Mission.IN_UTTER_DARKNESS.area)
+    }
+}
+
+lookup_id_to_mission: Dict[int, SC2Mission] = {
+    mission.id: mission for mission in SC2Mission
+}
+
+lookup_name_to_mission: Dict[str, SC2Mission] = {
+    mission.mission_name: mission for mission in SC2Mission
+}
+
+lookup_id_to_campaign: Dict[int, SC2Campaign] = {
+    campaign.id: campaign for campaign in SC2Campaign
 }
 
-lookup_id_to_mission: Dict[int, str] = {
-    data.id: mission_name for mission_name, data in vanilla_mission_req_table.items() if data.id}
+
+def get_campaign_missions(campaign: SC2Campaign) -> List[SC2Mission]:
+    return [mission for mission in SC2Mission if mission.campaign == campaign]
+
+
+def get_campaign_difficulty(campaign: SC2Campaign, excluded_missions: Set[SC2Mission] = None) -> MissionPools:
+    """
+
+    :param campaign:
+    :param excluded_missions:
+    :return: Campaign's the most difficult non-excluded mission
+    """
+    if excluded_missions is None:
+        excluded_missions = []
+    excluded_mission_set = set(excluded_missions)
+    included_missions = set(get_campaign_missions(campaign)).difference(excluded_mission_set)
+    return max([mission.pool for mission in included_missions])
+
+
+def get_campaign_goal_priority(campaign: SC2Campaign, excluded_missions: Set[SC2Mission] | frozenset[str] = None) -> SC2CampaignGoalPriority:
+    """
+    Gets a modified campaign goal priority.
+    If all the campaign's goal missions are excluded, it's ineligible to have the goal
+    If the campaign's very hard missions are excluded, the priority is lowered to hard
+    :param campaign:
+    :param excluded_missions:
+    :return:
+    """
+    if excluded_missions is None:
+        return campaign.goal_priority
+    else:
+        goal_mission_names = set([mission.mission_name for mission in get_campaign_potential_goal_missions(campaign)])
+        excluded_mission_set = set(excluded_missions)
+        remaining_goals = goal_mission_names.difference(excluded_mission_set)
```

Comment:
This bit still uses names, currently it compares strings with SC2Missions.

The function definition also still has a `frozenset[str]` in it.

Reply:
Fixed